pulumi/aws/anthos/__main__.py

```diff
@@ -90,8 +91,13 @@ def add_namespace(obj):
 if not accounts_db:
     accounts_db = 'postgresdb'
 
-accounts_db_uri = 'postgresql://' + str(accounts_admin) + ':' + str(accounts_pwd) + '@accounts-db:5432/' + str(
-    accounts_db)
+# The database password is a secret, and in order to use it in a string concat
+# we need to decrypt the password with Output.unsecret() before we use it.
+# This function provides the logic to accomplish this.
+accounts_db_uri = pulumi.Output.unsecret(accounts_pwd).apply(
```

Comment:
Consider modifying this such that Pulumi continues to use secrets:

```python
def create_pg_uri(password_object):
    user = str(accounts_admin)
    password = str(password_object)
    database = str(accounts_db)
    uri = f'postgresql://{user}:{password}@accounts-db:5432/{database}'
    return pulumi.Output.secret(uri)


accounts_db_uri = pulumi.Output.unsecret(accounts_pwd).apply(create_pg_uri)
```

Reply:
Fixed; that's a great example construct for this use case.